### PR TITLE
Mds 385

### PR DESCRIPTION
**Pull request by ajchilts**
Postman testing guide:

NB: WILL NEED TO RUN LATEST VERSION OF IF-STUB

In a browser

Visit http://localhost:9949/auth-login-stub/gg-sign-in
Redirect URL: http://localhost:9949/auth-login-stub/session
Enrolment Keys:
read:individuals-matching
read:individuals-income-laa-c1 (important so that fields query matches payload in IF)

Submit and copy authToken

Using Postman

Get matchId

POST http://localhost:9652
Authorization: Bearer token (paste in authToken from above)
Headers:
Content-Type: application/json
Accept: application/vnd.hmrc.P1.0+json
Body:
{
"firstName": "John",
"lastName": "Densmore",
"nino": "CS700100A",
"dateOfBirth": "1969-03-05"
}

Copy matchId from:
“/individuals/matching/<matchId>”

Income Root

GET http://localhost:9652/
Authorization: Bearer token (paste in authToken from above)
Headers:
Accept: application/vnd.hmrc.P2.0+json
Query Parameters:
matchId: <paste matchId from above>

Income PAYE

GET http://localhost:9652/paye
Authorization: Bearer token (paste in authToken from above)
Headers:
Accept: application/vnd.hmrc.P2.0+json
Query parameters:
matchId: <pasteMatchId from above>
fromDate: 2019-01-01
toDate: 2020-03-01

Expected result:

2x Paye Income

Stubs branch: https://github.com/hmrc/if-stub/pull/10

### CODE REVIEW

File: conf/application.conf [hunk 1]
```diff
@@ -311,6 +305,7 @@ api-config {
 
     endpoints {
         incomeSaSource {
+          title = "Get an individual's SA source data"
             endpoint = "/individuals/income/sa/source?matchId=<matchId>{&startYear,endYear}"
```
Comment (owen-opencast) [inline conf/application.conf:309]: Fix text alignment

Comment (owen-opencast) [inline conf/application.conf:309]: I think startYear/Date and endYear/Date need updated

File: test/component/uk/gov/hmrc/individualsincomeapi/VersioningSpec.scala [hunk 2]
```diff
@@ -112,6 +112,63 @@ class VersioningSpec extends BaseSpec {
              }
          }""")
 
+  private def validResponsePayloadP2 =
+    Json.parse(s"""
+         {
+            "_links":{
+              "incomeSaUkProperties":{
+                "href":"individuals/income/sa/uk-properties?matchId=57072660-1df9-4aeb-b4ea-cd2d7f96e430{&startYear,endYear}",
```
Comment (owen-opencast) [inline test/component/uk/gov/hmrc/individualsincomeapi/VersioningSpec.scala:120]: startYear/Date and endYear/Date

File: test/component/uk/gov/hmrc/individualsincomeapi/controllers/v2/IndividualIncomeSpec.scala [hunk 3]
```diff
@@ -18,20 +18,32 @@ package component.uk.gov.hmrc.individualsincomeapi.controllers.v2
...
+import play.api.libs.json.Json
 import play.api.test.Helpers._
 import scalaj.http.Http
-import uk.gov.hmrc.individualsincomeapi.domain.SandboxIncomeData.sandboxMatchId
+import uk.gov.hmrc.individualsincomeapi.domain.integrationframework.IfPaye
+import uk.gov.hmrc.individualsincomeapi.domain.v1.SandboxIncomeData.sandboxMatchId
+import utils.IncomePayeHelpers
 
-class IndividualIncomeSpec extends BaseSpec {
+class IndividualIncomeSpec extends BaseSpec with IncomePayeHelpers {
```
Comment (owen-opencast) [inline test/component/uk/gov/hmrc/individualsincomeapi/controllers/v2/IndividualIncomeSpec.scala:29]: I can't view the docs on dev hub at the moment, but in employments I added checks for the error messages around fromDates / toDates at @parkeradam's suggestion -  to being before from etc. Might be worth adding to this test.

**Review by owen-opencast — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/individualsincomeapi/domain/integrationframework/IfPayeEntry.scala [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+
+  val payFrequencyValues = Seq("W1", "W2", "W4", "M1", "M3", "M6", "MA", "IO", "IR")
+
+  def isInPayFrequency(implicit rds: Reads[String]): Reads[String] =
+    verifying(value => payFrequencyValues.contains(value))
+
+  implicit val grossEarningsForNicsFormat: Format[IfGrossEarningsForNics] = Format(
+    (
+      (JsPath \ "inPayPeriod1").readNullable[Double](verifying(paymentAmountValidator)) and
+        (JsPath \ "inPayPeriod2").readNullable[Double](verifying(paymentAmountValidator)) and
```
[the remark this answers is not in the document]
Comment (ajchilts) [reply app/uk/gov/hmrc/individualsincomeapi/domain/integrationframework/IfPayeEntry.scala:137]: IntelliJ seems to be making this decision on my behalf :-|

**Review by owen-opencast — APPROVED** [review]

**Pull request merged**